### PR TITLE
Add smoke test to mask sensitive data in log body

### PR DESCRIPTION
**Pull request by heyams**
https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-telemetry-processors-examples#masking-sensitive-data-in-log-message sample uses the incorrect regex. Fix is [here](https://github.com/MicrosoftDocs/azure-docs-pr/pull/259827)

### CODE REVIEW

**Review by jeanbisutti — APPROVED** [review]

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "percentage": 100
+  },
+  "preview": {
+    "processors": [
+      {
+        "type": "log",
+        "body": {
+          "toAttributes": {
+            "rules": [
+              "^User account with userId (?<redactedUserId>[\\da-zA-Z]+)[\\w\\s]+"
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:16]: I didn't follow what the need for the trailing portion is?

```suggestion
              "^User account with userId (?<redactedUserId>[\\da-zA-Z]+)"
```

Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:16]: this wouldn't work. there is text after user id.

The original string is "User account with userId 123456xx failed to login"

Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:16]: "User account with userId {redactedUserId} failed to login"

Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:16]: @trask i tried it, this doesn't work

Comment (trask) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:16]: I just tried it on your fork and it worked for me, I'll commit the suggestion and see what happens in CI

Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json]: it worked for me too, not sure why it didn't work yesterday on my local.

Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json]: https://github.com/MicrosoftDocs/azure-docs-pr/pull/260040

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          }
+        ],
+        "id": "/attribute/deleteRedactedUserId"
+      },
+      {
+        "type": "log",
+        "body": {
+          "toAttributes": {
+            "rules": [
+              "This is my \"email\" : \"(?<redactedEmail>.*)\""
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:37]: this is probably more typical (and safer)
```suggestion
              "This is my \"email\" : \"(?<redactedEmail>[^\"]*)\""
```

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "percentage": 100
+  },
+  "preview": {
+    "processors": [
+      {
+        "type": "log",
+        "body": {
+          "toAttributes": {
+            "rules": [
+              "userId (?<redactedUserId>[\\da-zA-Z]+)"
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-mask-log-body.json:16]: just a style preference in this particular case

```suggestion
              "userId (?<redactedUserId>[0-9a-zA-Z]+)"
```

**Review by trask — APPROVED** [review]
👍

**Review by trask — APPROVED** [review]

**Pull request merged**